akka-stream/src/main/scala/akka/stream/impl/SynchronousFileSubscriber.scala

```diff
@@ -0,0 +1,76 @@
+/**
+ * Copyright (C) 2015 Typesafe Inc. <http://www.typesafe.com>
+ */
+package akka.stream.impl
+
+import java.io.{ File, RandomAccessFile }
+import java.nio.channels.FileChannel
+
+import akka.actor.{ ActorLogging, Props }
+import akka.stream.actor.{ ActorSubscriberMessage, OneByOneRequestStrategy }
+import akka.util.ByteString
+
+import scala.concurrent.Promise
+
+/** INTERNAL API */
+private[akka] object SynchronousFileSubscriber {
+  def props(f: File, completionPromise: Promise[Long], append: Boolean) =
+    Props(classOf[SynchronousFileSubscriber], f, completionPromise, append)
+
+}
+
+/** INTERNAL API */
+private[akka] class SynchronousFileSubscriber(f: File, completionPromise: Promise[Long], append: Boolean) extends akka.stream.actor.ActorSubscriber
+  with ActorLogging {
+
+  // TODO since ActorPublisher is typed, shouldn't ActorSubscriber be as well?
+
+  override protected val requestStrategy = OneByOneRequestStrategy // TODO better one here ?
```

Comment:
WatermarkRequestStrategy with highWatermark == maxBufferSize from the OperationAttributes?